roles/openshift_facts/library/openshift_facts.py

```diff
@@ -430,6 +408,46 @@ def set_aggregate_facts(facts):
 
     return facts
 
+def set_deployment_facts_if_unset(facts):
```

Comment:
@sdodson I'm not against breaking it out into multiple functions if we need to work around pylint....

At some point we will probably need to refactor openshift_facts to clean up the way we are doing things a bit...

One approach that might simplify the way openshift_facts works would be to do the following:
- define a dictionary for each role (common, master, node, etc) which lists all of the default values for that role
- read in the local_facts file
- merge the local_facts from the file with the ones passed into the openshift_facts module (local_facts passed in win)
- create the initial facts dictionary based on the roles contained in the local_facts and merge the defaults with the local_facts from the default dictionaries. (local_facts win)
- set any facts that may need to be computed at runtime
  - set_url_facts_if_unset
  - set_aggregate_facts
  - possibly others...